### PR TITLE
Wanderer Flagship 

### PR DESCRIPTION
**Pull request by chadenr**
Added Wanderer Flagship based on Bactrian and Derecho. Very large ship with high cargo and bunk capacity and the ability to field Fly Catchers. Unlocks with Hurricane.

### CODE REVIEW

**Comment by chadenr** [comment]
> Thanks for the contribution, but I think this would be better off as a plugin rather than an addition to the base game--at least until the following issues are resolved:
> 
> 1. There's no storyline changes to explain this enormous new ship
> 2. There are no mission balance changes to account for it3. The ship's sprite clearly appears to be a Hurricane and a Derecho jammed together, rather than a typically elegant Wanderer design
> 3. The ship's sprite has no texturing
> 4. The ship's role is unclear and unrealistic. It's far too heavily armed to be a science or passenger ship, but not heavily-armed enough to be a "flagship" of any sort. No military would send a ship like this into dangerous territory un-escorted; it would be too valuable to risk. Instead it would have a whole fleet to escort it, which means that it would not need anywhere near its current level of armament and shielding.

Thank you for the detailed reply.
* Am working on story changes and am aware that other contributors are working on the Wanderer storyline as well. I don't want to step on their toes.
* No balance changes are required. This is a tier upgrade to an existing ship. It's less powerful than the existing best warship and hauls less cargo than the existing best cargo ship. 
* If someone wants to contribute a sprite, they are welcome to. I am not a graphic artist.
* The ship is a tier upgrade to an existing ship with stats directly and linearly correlated to that ship.

File: data/wanderer dire storm.txt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+			"Tempest (Missile)"
+			"Dire Storm"
+			"Flycatcher" 4
+	shipyard "Wanderer Advanced"
+		"Hurricane"
+		"Dire Storm"
+	planet "Spera Anatrusk"
+		shipyard "Wanderer Basics"
+		
+ship "Bactrian"
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: Why is this here?

File: data/wanderer dire storm.txt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	sprite "ship/dire storm"
+	attributes
+		category "Heavy Warship"
+		licenses
+			"Wanderer Military"
+		"cost" 40500000
+		"shields" 54128
+		"hull" 48962
+		"required crew" 72
+		"bunks" 237
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: That's a lot of bunks. Is this a passenger ship?

File: data/wanderer dire storm.txt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		"mass" 832
+		"drag" 13.9
+		"heat dissipation" 0.5
+		"fuel capacity" 816
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: Why the non-round number?

File: data/wanderer dire storm.txt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+		"cargo space" 424
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: That's a huge amount of cargo space. Is this a cargo ship?

File: data/wanderer dire storm.txt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+		"outfit space" 864
+		"weapon capacity" 276
+		"engine capacity" 240
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: That's quite a lot of outfit, weapon, and engine space. Is this a warship?

File: data/wanderer dire storm.txt [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+	drone 84 63
+	drone -84 63
+	drone -72 63
+	drone 72 63
+	explode "small explosion" 40
+	explode "medium explosion" 70
+	explode "large explosion" 50
+	explode "huge explosion" 20
+	"final explode" "final explosion large" 1
+	description "The Dire Storm is a deep space command, control and science vessel, designed to be one of the first ships through the Eye, to begin cataloging the dangers of the unknown space beyond, and to provide a secure, mobile headquarters for the warships that stay behind to cover the Wanderers' retreat. Endless banks of climate-controlled storage allow the dozen (or so) science labs to take hundreds of soil, water and even biological specimens from desolate worlds. Four drones allow remote, efficient harvesting and in an emergency can provide cover for a fleet of ships."
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: "command, control and science vessel" == not a heavy warship
"designed to be one of the first ships through the Eye" == either a fast scouting vessel or a heavy warship

File: data/wanderer dire storm.txt [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	
+ship "Dire Storm" "Dire Storm (Passenger)"
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: We already have a PR to add a much more appropriate Wanderer passenger ship: https://github.com/endless-sky/endless-sky/pull/4102

File: data/wanderer dire storm.txt [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+	gun "Thunderhead Launcher"
+	gun "Thunderhead Launcher"
+	gun "Thunderhead Launcher"
+	turret "Dual Sunbeam Turret"
+	turret
+	turret "Wanderer Anti-Missile"
+	turret "Wanderer Anti-Missile"
+	
+event "wanderers: hurricane mass production"
+	fleet "Wanderer Defense"
```
Comment (Pointedstick) [inline data/wanderer dire storm.txt]: If this is a science ship with dozens of climate controlled labs, why is it a part of planetary defense fleets?

File: data/wanderer dire storm.txt (hunk 3 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: Because wanderer ships have ram scoops stock.

File: data/wanderer dire storm.txt (hunk 1 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: Because wanderers have neither missions nor outfitters nor shipyards past the wormhole, and I'm working on that. Arguably could go into a different PR but if I need to put the event here anyway to unlock the dire storm it makes sense to have everything in one spot.

File: data/wanderer dire storm.txt (hunk 6 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: > Three more things, in addition to the inline comments added below:
> 
> 1. Why is this in its own file? That would be appropriate for a plug-in, but for content PR'd to the base game, a new ship should go into the existing `wanderer ships.txt` file.
People in discord told me the standard was to have a separate file for this sort of thing. We should maybe agree on the standard for that sort of thing.
> 2. When you submit a PR to add game content, it needs to be complete. You can't just say, "I'm not an artist, someone else can add art." If you're not capable of producing the appropriate art assets, then you probably shouldn't submit the PR until you gain that ability and do it yourself, or arrange ahead of time for someone else to do it so you can submit those assets in the PR.
This is after collaborating on discord with someone for the art. Surveillance/Botspam and I have already done a couple iterations on my original art.
> 3. It's considered a best practice to only only open a PR once everything is done and tested. The review process is optimized for helping you polish up an already solid offering and preventing errors from seeping in; it's not the ideal way to collaboratively generate new content on a continuous basis. The Discord server is a better place to do that.
If you are waiting for content to be perfect and complete before pulling it, that might explain why this project has been in dev hell for two years.

File: data/wanderer dire storm.txt (hunk 1 above)
Comment (Pointedstick) [reply data/wanderer dire storm.txt]: ...What? I'm not sure what the Bactrian has to do with this PR in the first place.

File: data/wanderer dire storm.txt (hunk 7 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: The Centipede and the Riptide are not tier upgrades on the Bactrian.

File: data/wanderer dire storm.txt (hunk 8 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: It's a command and control vessel. Think Enterprise.

File: data/wanderer dire storm.txt (hunk 2 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: It's a deep space exploration cruiser. To boldly go where no bird has gone before.

File: data/wanderer dire storm.txt (hunk 4 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: Spam.

Comment (Pointedstick) [reply data/wanderer dire storm.txt]: ...What?

File: data/wanderer dire storm.txt (hunk 7 above)
Comment (Pointedstick) [reply data/wanderer dire storm.txt]: What does the Bactrian have to do with anything in this PR?

File: data/wanderer dire storm.txt (hunk 6 above)
Comment (Pointedstick) [reply data/wanderer dire storm.txt]: It doesn't need to be perfect, but it does need to be _ready to review_. This PR isn't ready to review.

File: data/wanderer dire storm.txt (hunk 2 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: They literally have the need to explore dangerous, unexplored parts of the galaxy every time they go through the Eye. Their need to explore the new frontier literally forms the basis of the wanderer campaign, first defending their old worlds long enough for the eye to open, and then taming and settling the wild reaches. And their science doesn't take place exclusively in safe, sanitary laboratories; it literally takes place around and with life forms that can and will eat them if given the chance.

File: data/wanderer dire storm.txt (hunk 8 above)
Comment (Pointedstick) [reply data/wanderer dire storm.txt]: I'm as big a Star Trek fan as they come, but the Enterprise doesn't really make that much logical sense even in its own universe. It's a flagship that usually operates alone. It's the heaviest warship in the Federation fleet that's often called upon to go into battle, yet it houses a significant civilian population as well as a large amount of staff and equipment for scientific endeavors, all of whom are put in great and unnecessary risk every time the ship goes into battle.

Realistically, this kind of jack-of-all-trades vessel makes no sense. It makes far more sense to have more than one ship, each optimized for its own role. For example, to do science, you would want a dedicated science ship. To _briefly_ do science in a hostile location, you would want your science ship protected with a fleet of dedicated military warships so it can _quickly_ do the science and then get out of there ASAP. Or you would want the science ship to be incredibly light and fast. Like, say, a Scout or a Bounder.

The only reason why the jack-of-all-trades Bactrian exists in Endless Sky is because it hearkens back from a brief time when a self-sufficient warship/freighter/transport _did_ make sense. But it's a relic--a product of an older time marked by exploration and colonization in regions of space with no military support. That doesn't describe the Wanderers' situation.

File: data/wanderer dire storm.txt (hunk 2 above)
Comment (Pointedstick) [reply data/wanderer dire storm.txt]: Yeah, and that's why they have a strong military fleet. They don't need to press their scientific staff into military service. They can just protect them with real warships.

Comment (Amazinite) [reply data/wanderer dire storm.txt]: The Wanderers only seem to begin building new ships as they need them. If your argument for the Dire Storm is that the Wanderers need to explore every time the Eye is opened, then they don't need the DIre Storm here because of how the story already has the player doing the exploration for the Wanderers.

File: data/wanderer dire storm.txt (hunk 8 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: The Wanderers don't make plans to briefly do anything. They go into hostile territory, and they settle there for thousands of years. They have no military experience and no interest in getting military experience, but they do plan to survive in harsh conditions that literally killed the previous inhabitants of the sector. 

If you don't think the wanderers are explorers, scientists and settlers that's totally fine, but let's agree to differ on this point. You don't think the ship is needed. Thank you for your opinion.

**Comment by Pointedstick** [comment]
After much discussion, we finally got to the "problem statement" that this patch aims to solve:

> Bactrian is presently best flagship and presents a strong mix of cargo, bunks and weaponry that no other ship in the entire game can match, including ships supposedly several tiers ahead of Bactrian

So you're trying to create a ship that's a higher-tier alien version of the Bactrian.

But this game doesn't _need_ a better Bactrian. It doesn't need the existing Bactrian "nerfed" either. The Bactrian is fine. Compared to higher-tier ships, the Bactrian is slower, far less shielded and armored, and has vastly less weapon and outfit capacity compared to higher tier flagships like the Hurricane or Kar Ik Vot. It is by no means the "best flagship." Yes, its cargo and bunks are indeed quite good, but so what? That's its thing, it's _raison d'être_. It melts in a real fight with higher tier ships, or even a reasonable fleet of Human ships. Just the other day my wife trashed a Bactrian escorted by a Carrier in one of the Wolf Pack missions while flying a lone Osprey armed with heavy rocket launchers. It's not the best flagship in the game; it's a good ship that has a strong niche (salvaging and capturing). And that's just fine. :)

File: data/wanderer dire storm.txt (hunk 8 above)
Comment (Pointedstick) [reply data/wanderer dire storm.txt]: The wanderers certainly are explorers, scientists and settlers. My contention is that this multipurpose vessel does not make logical sense.

You're misunderstanding why self-sufficient jack-of-all-trade ships exist; the only logical reason is because there are no armed escorts available. Any time escorts are available, it makes more sense to have each ship perform a dedicated specialist role rather than try to generalize. To settle in potentially hostile lands, you'd have an unarmed yet well defended colony ship escorted by a military task force to protect it. The only reason to put a ton of guns and engines on the colony ship is if there was no military escort available--which was the case for the Bactrian. But it is not the case for the Wanderers as they pass through the Eye. They have plenty of warships available to protect their vulnerable transports and cargo ships.

**Comment by chadenr** [comment]
> After much discussion, we finally got to the "problem statement" that this patch aims to solve:
> 
> > Bactrian is presently best flagship and presents a strong mix of cargo, bunks and weaponry that no other ship in the entire game can match, including ships supposedly several tiers ahead of Bactrian
> 
> So you're trying to create a ship that's a higher-tier alien version of the Bactrian.
> 
> But this game doesn't _need_ a better Bactrian. It doesn't need the existing Bactrian "nerfed" either. The Bactrian is fine. Compared to higher-tier ships, the Bactrian is slower, far less shielded and armored, and has vastly less weapon and outfit capacity compared to higher tier flagships like the Hurricane or Kar Ik Vot. It is by no means the "best flagship." Yes, its cargo and bunks are indeed quite good, but so what? That's its thing, it's _raison d'être_. It melts in a real fight with higher tier ships, or even a reasonable fleet of Human ships. Just the other day my wife trashed a Bactrian escorted by a Carrier in one of the Wolf Pack missions while flying a lone Osprey armed with heavy rocket launchers. It's not the best flagship in the game; it's a good ship that has a strong niche (salvaging and capturing). And that's just fine. :)



> After much discussion, we finally got to the "problem statement" that this patch aims to solve:
> 
> > Bactrian is presently best flagship and presents a strong mix of cargo, bunks and weaponry that no other ship in the entire game can match, including ships supposedly several tiers ahead of Bactrian
> 
> So you're trying to create a ship that's a higher-tier alien version of the Bactrian.
> 
> But this game doesn't _need_ a better Bactrian. It doesn't need the existing Bactrian "nerfed" either. The Bactrian is fine. Compared to higher-tier ships, the Bactrian is slower, far less shielded and armored, and has vastly less weapon and outfit capacity compared to higher tier flagships like the Hurricane or Kar Ik Vot. It is by no means the "best flagship." Yes, its cargo and bunks are indeed quite good, but so what? That's its thing, it's _raison d'être_. It melts in a real fight with higher tier ships, or even a reasonable fleet of Human ships. Just the other day my wife trashed a Bactrian escorted by a Carrier in one of the Wolf Pack missions while flying a lone Osprey armed with heavy rocket launchers. It's not the best flagship in the game; it's a good ship that has a strong niche (salvaging and capturing). And that's just fine. :)

Among its peers, Bactrian is a straight upgrade over Leviathan and Falcon in terms of a straight fight playing to the warship's strengths. Bactrian has more shields, more outfit space and more guns, meaning that if it's intended to be a "tier 1" ship, it does need a nerf because it's better than any other tier 1 ship.

My understanding of the ship tier system is that tier 1 ships are supposed to eventually become obsolete, but Bactrian is never obsoleted because there is no equivalent higher-tier ship. For example, a player flying a Leviathan or a Navy Cruiser will eventually want to upgrade to a Shield Beetle or a Derecho. A player flying a Bactrian has no equivalent upgrade. At its tier, the Bactrian is the best ship. At higher tiers, the Bactrian continues to be the best ship for salvaging, looting and capturing, which the player (and only the player) will continue to do throughout the game.

File: data/wanderer dire storm.txt (hunk 3 above)
Comment (jafdy) [reply data/wanderer dire storm.txt]: That doesn't answer the question.

File: data/wanderer dire storm.txt (hunk 7 above)
Comment (jafdy) [reply data/wanderer dire storm.txt]: We have long agreed that the Bactrain is too powerful even in T2, I don't see why adding a T2 version will help that.

**Comment by jafdy** [comment]
Unless crew sizes are increased drastically with tier then the Bactrian will always remain just as good in that role, as long as it escorted by an increasingly large and powerful war fleet that can compensate for it's (relatively) week fighting strength. 
Currently players are forced to make a decision between keeping that capturing/plundering power and being able to fight worth a damn. This is an interesting decision, and one that does not currently exist at T1. By adding this you introduce the problems the Bactrian caused in T1 to T2 thus removing that interesting decision until T3, where you will presumably try to do the same.

I'm not against say, an "armoured" (read shielded) transport of some kind, designed to not die while it's attended war fleet kills any attackers but creating a jack-of-all-trades doesn't help making interesting decisions.

**Comment by chadenr** [comment]
> Unless crew sizes are increased drastically with tier then the Bactrian will always remain just as good in that role, as long as it escorted by an increasingly large and powerful war fleet that can compensate for it's (relatively) week fighting strength.
> Currently players are forced to make a decision between keeping that capturing/plundering power and being able to fight worth a damn. This is an interesting decision, and one that does not currently exist at T1. By adding this you introduce the problems the Bactrian caused in T1 to T2 thus removing that interesting decision until T3, where you will presumably try to do the same.
> 
> I'm not against say, an "armoured" (read shielded) transport of some kind, designed to not die while it's attended war fleet kills any attackers but creating a jack-of-all-trades doesn't help making interesting decisions.

By the time hurricane and dire storm are unlocked, the ability to capture ships has stopped being an interesting choice because none of the enemy ships have crew. At that point, the Bactrian is an armored cargo looter, with more combined shields and maneuverability than other choices like the Deep River or the . In the event that at some time in the future, capturing again becomes an interesting choice, you won't lose anything by upgrading to a dire storm, but by nerfing Bactrian cargo I am introducing other interesting choices regarding the ability to hold and capture loot, like the Water Bug and the Sojourner. I am specifically setting Bactrian cargo equivalent to Sojourner cargo and reducing its shields below the Sojourner's to produce an interesting choice between the two, where Bactrian is able to capture high-end heavy warships and Sojourner has more agility and a lighter footprint.

File: data/wanderer dire storm.txt (hunk 3 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: Because wanderer ships have stock ram scoops, they benefit from a non-round fuel number because it takes them less time to charge their first ram-fueled jump. If a ship has fuel for N jumps, it can jump its N+1th jump faster, in the same way that a ship with a ramscoop and a scram drive benefits from having 700 fuel over 600 fuel.

File: data/wanderer dire storm.txt (hunk 2 above)
Comment (chadenr) [reply data/wanderer dire storm.txt]: Player has successfully not died after jumping into a number of systems. However, if the wanderers intend to settle there, their civilians need better protection than is the norm for their species because a random fleet of rabid drones -- possibly led by an alpha or a korath raiding fleet -- could literally show up at any minute. Even after the wanderer mid campaign is resolved, they have good reason to protect their civilian science efforts and are culturally not well equipped to have war fleets on standby.

**Comment by Amazinite** [comment]
This PR is in no position to be merged. As Pointed stated, it's not even really in a review-ready state. The changes aren't added to their respective files (instead being added as if it was a plugin), the art is unfinished, the reasoning behind the ship is questionable, and the ship is just added into the middle of the Wanderer story without explanation then is never used by the Wanderers in any missions.

If you're looking to create a massive warship for the Wanderers, I think the reasoning behind it should be altered and the art of the ship should look more like the Tele'ekite Wanderer ships than the old Wanderer ships. This should probably go back to the drawing board before it's in a PR-ready state. 

Alternatively, you could finish the art for the ship then simply add the ship to the Wanderer ships file and leave it up to MZ to use like #4102. Just a heads up though, keep PRs to a single topic. None of this "Here's my ship, also I nerfed the Bactrian." (And on that topic, nerfing the Bactrian to add this ship is the perfect example of why the Bactrian is a problem in the first place.)

**Comment by Nescio0** [comment]
Actually your sprite looks promising. What you have to do (cf. #4102):
* do not include unrelated changes (e.g. Bactrian, fleets, etc.)
* create a thumbnail
* provide @2x images
* include all assets (Blender, GIMP, other files) in your opening post
* insert yourself into the copyright file
